### PR TITLE
fake nav bar branch 

### PR DESCRIPTION
**Pull request by TOMalley104**
these will get better.

### CODE REVIEW

File: spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+//
+//  Created by Tom O'Malley on 5/6/16.
+//  Copyright © 2016 intrepid. All rights reserved.
+//
+
+import Foundation
+
+class ArtistHeaderView : UIView {
+
+    @IBOutlet var view: UIView!
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift]: dont need var view here I think

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift [hunk 2]
```diff
@@ -8,47 +8,42 @@
...
-        self.navigationController?.navigationBar.setBackgroundImage(UIImage(), forBarMetrics: UIBarMetrics.Default)
-        self.navigationController?.navigationBar.shadowImage = UIImage()
-        self.navigationController?.navigationBar.translucent = true
+        self.setupMoreBarButton()
+    }
+
+    // MARK: Actions
+
+    func moreTapped() {
+        print("will this ever do anything?")
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: i'm not sure, probably should remove it ofr this pr : )

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift [hunk 3]
```diff
@@ -88,7 +83,46 @@ class ArtistViewController: UIViewController, UIScrollViewDelegate, UITableViewD
         return 40;
     }
 
-    // MARK: Helpers
+    // MARK: View Helpers
+
+    var headerIsOnTop = false
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: typically we define all our vars on top, in one place.

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift [hunk 4]
```diff
@@ -88,7 +83,46 @@ class ArtistViewController: UIViewController, UIScrollViewDelegate, UITableViewD
...
+        if let scrollViewIndex = self.view.subviews.indexOf(self.scrollView),
+            let artistHeaderViewIndex = self.view.subviews.indexOf(self.artistHeaderView) {
+            self.view.exchangeSubviewAtIndex(scrollViewIndex, withSubviewAtIndex: artistHeaderViewIndex)
+        }
+    }
+
+    // MARK: View Setup
+
+    func setupMoreBarButton() {
+        let iconSize = 40 as CGFloat
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: 40.0

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+class ArtistSearchTableViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, UISearchBarDelegate, SPTAuthViewDelegate {
+
+    @IBOutlet weak var searchBar: UISearchBar!
+    @IBOutlet weak var logInBarButton: UIBarButtonItem!
+    @IBOutlet weak var tableView: UITableView!
+
+    var session: SPTSession? {
+        didSet {
+            if let session = self.session {
+                self.logInBarButton.enabled = !session.isValid() // FIXME: isValid() is lying
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: can you elaborate more on this pr? What needs to be fixed? When leaving TODO, try to think about how you can communicate to your teammate what needs to be done

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+            } else if let listPage = listPage as? SPTListPage,
+                let items = listPage.items as? [SPTPartialArtist] {
+                self.searchResults = items
+                self.tableView.reloadData()
+                let artistURIs = items.map { $0.uri }
+                SPTArtist.artistsWithURIs(artistURIs, session: self.session, callback: { error, object in
+                    if let error = error {
+                        print("fetch SPTArtist error: \(error.localizedDescription)") // FIXME: throw application error
+                    } else if let artists = object as? [SPTArtist] {
+                        self.fullArtists = artists
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: You might have a bug here when the network conditions are slow. In this case, you might get a crash when accessing fullArtists.

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            self.session = session
+        }
+    }
+
+    // MARK: Notification Handling
+
+    func recieveSpotifyLogInNotification(notification:NSNotification) {
+        self.dismissViewControllerAnimated(false, completion: nil)
+        if let session = notification.userInfo?["session"] as? SPTSession {
+            self.session = session; print("session recieved")
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: remove print session

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+            print("we have a session already! it is... \(session.isValid() ? "VALID" : "INVALID")") // FIXME: isValid() is lying
+            self.session = session
+        }
+    }
+
+    // MARK: Notification Handling
+
+    func recieveSpotifyLogInNotification(notification:NSNotification) {
+        self.dismissViewControllerAnimated(false, completion: nil)
+        if let session = notification.userInfo?["session"] as? SPTSession {
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: time to constantize session, it has been used several times

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+    func authenticationViewController(authenticationViewController: SPTAuthViewController!, didFailToLogin error: NSError!) {
+        // FIXME: make this an application error, handle it
+    }
+
+    func authenticationViewControllerDidCancelLogin(authenticationViewController: SPTAuthViewController!) {
+        // this method is required, but i have no use for it ...
+    }
+
+    func authenticationViewController(authenticationViewController: SPTAuthViewController!, didLoginWithSession session: SPTSession!) {
+        print("login success, session: \(session)")
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: remove print statements

File: spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift (hunk 1 above)
Comment (TOMalley104) [inline spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift]: its part of how i learned how to use xibs, would enjoy learning how to do so without it!

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift (hunk 2 above)
Comment (TOMalley104) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: I needed a selector to give the bar button when i created it... can i pass nil there? or would you recommend just making a bar button in storyboard and setting the new image to it?

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        self.session = session
+    }
+
+    // MARK: Navigation 
+
+    override func shouldPerformSegueWithIdentifier(identifier: String, sender: AnyObject?) -> Bool {
+        if let cell = sender as? UITableViewCell,
+            let ip = self.tableView.indexPathForCell(cell) {
+            return self.fullArtists.indices.contains(ip.row)
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: I see, this prevents you from blowing up when fullArtists have not returned, but I think you should be able to still push to the next page even if the fullartists have not returned.

Comment (TOMalley104) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: so SPTPartialArtist is pretty useless.. it basically only gives us the artist's name and URI (for fetching the SPTArtist version). One of two refactors:

1. make this page only show SPTArtists, meaning it wont show anything unless the full artist fetch is successful. 
2. give the ArtistViewController the SPTPartialArtists and make it responsible for fetching the full artist. 

I'm more inclined to do 1 but am interested in your suggestion

File: spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift (hunk 1 above)
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift]: did you fix this?

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift (hunk 10 above)
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: so lets leave this alone for now and do it later.

**Pull request merged**